src/System.Private.CoreLib/src/System/Utf8String.Searching.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Unicode;
+
+namespace System
+{
+    public sealed partial class Utf8String
+    {
+        // Ordinal search
+        public bool Contains(char value)
+        {
+            return Rune.TryCreate(value, out Rune result) && Contains(result);
+        }
+
+        // Ordinal search
+        public bool Contains(Rune value)
+        {
+            // TODO_UTF8STRING: This should be split into two methods:
+            // One which operates on a single-byte (ASCII) search value,
+            // the other which operates on a multi-byte (non-ASCII) search value.
+
+            Span<byte> runeBytes = stackalloc byte[Utf8Utility.MaxBytesPerScalar];
```

Comment:
@gfoidl I was experimenting in a separate branch with moving some of this into the JIT if possible, using a technique very similar to what you had done. But it'll be a while before I get around to performing these sorts of optimizations. Pseudocode:

```cs
public bool Contains(Rune value)
{
    if (JitHelpers.TryConstRuneToAscii(value, out byte valueAscii)
    {
        // if the JIT knows that 'value' is const and ASCII, go down this code path
        // (valueAscii would've gotten baked into the codegen as a constant)
        return SpanHelpers.IndexOf(..., valueAscii) >= 0
    }
    else if (JitHelpers.TryConstRuneToUtf8(value, out int data, out int dataLength))
    {
        // if the JIT knows that 'value' is const and not ASCII, go down this code path
        // (data and dataLength would've gotten baked into the codegen as constants)
        return SpanHelpers.IndexOf(..., ref Unsafe.As<int, byte>(ref data), dataLength) >= 0;
    }
    else
    {
        // if the JIT can't reason about the const-ness of 'value', go down this code path
        // stackalloc and convert at runtime
    }
}
```

Reply:
(In the above example it would be perfectly valid for the `TryConstXyz` methods to be hardcoded to always return `false`, and the first two branches would be folded away by the JIT. The end result is that the code would still work; it just wouldn't be as optimized.)